tests/test_validate_parameters.py

```diff
@@ -0,0 +1,53 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of REANA.
+# Copyright (C) 2021 CERN.
+#
+# REANA is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""REANA client validate parameters tests."""
+
+import yaml
+
+from reana_client.validation import validate_serial_parameters
+
+
+def test_validate_parameters_serial(create_yaml_workflow_schema, capsys):
+    """Validate parameters for Serial workflows."""
+    reana_yaml = yaml.load(create_yaml_workflow_schema, Loader=yaml.FullLoader)
+    reana_yaml["inputs"]["parameters"]["foo"] = "foo"
+    validate_serial_parameters(reana_yaml)
+    captured = capsys.readouterr()
+    assert 'WARNING: Input parameter "foo" is not being used' in captured.out
+    del reana_yaml["inputs"]["parameters"]["foo"]
+
+    reana_yaml["workflow"]["specification"]["steps"].append(
+        {"commands": ["python ${foo} --bar"]}
+    )
+    validate_serial_parameters(reana_yaml)
+    captured = capsys.readouterr()
+    assert (
+        'WARNING: Parameter "foo" found on step "Unnamed" is not defined in inputs parameters'
+        in captured.out
+    )
+
+    reana_yaml["workflow"]["specification"]["steps"].append(
+        {"commands": ["python ${bar} --foo"], "name": "baz"}
+    )
+    validate_serial_parameters(reana_yaml)
+    captured = capsys.readouterr()
+    assert (
+        'WARNING: Parameter "bar" found on step "baz" is not defined in inputs parameters'
+        in captured.out
+    )
+
+    reana_yaml["workflow"]["specification"]["steps"].append(
+        {"commands": ["python ${bar} --foo"], "name": "qux"}
+    )
+    validate_serial_parameters(reana_yaml)
+    captured = capsys.readouterr()
+    assert (
+        'WARNING: Parameter "bar" found on steps "baz, qux" is not defined in inputs parameters'
+        in captured.out
+    )
```

Comment:
Perhaps we could add one last test to make sure that escaped environment variables are not cougth as false positives? (similar to second case listed [here](https://github.com/reanahub/reana-client/pull/469#pullrequestreview-602772503)).